File: MatlabCode/02_b_Homework_MonteCarlo/characteristic_curve.m
function [eh]=characteristic_curve(vh)

% cut-in, v-rated, cut-out wind speed [m/s]
Vi=3;
Vr=15;
Vo=25;

% max energy [kWh]
E=2000;

% energy (eh) for given wind speed (vh)
if vh<Vi
    eh=0;
elseif vh<Vr
    eh=E/(Vr-Vi)^3*(vh-Vi)^3;
elseif vh<Vo
    eh=E;
else
    eh=0;
end

File: PythonCode/02_b_Homework_MonteCarlo/characteristic_curve.py
import numpy as np


def characteristic_curve(vh):
    vi = 3                  # cut-in speed
    vr = 15                 # rated speed
    vo = 25                 # cut-out speed
    E = 2000                # max generated energy [kWh]

    # energy during hour h (eh) for given wind speed (vh)
    if vh < vh:
        eh = 0
    elif vh < vr:
        eh = (E * (vh - vi)**3)/((vr - vi)**3)
    elif vh < vo:
        eh = E
    else:
        eh = 0

    eh = np.asarray(eh)
    return eh
